diff --git a/include/target.mk b/include/target.mk
index f129298..cacf6b9 100644
--- a/include/target.mk
+++ b/include/target.mk
@@ -208,7 +208,7 @@ ifeq ($(DUMP),1)
     else
       CPU_TYPE ?= mips32
     endif
-    CPU_CFLAGS += -mno-branch-likely
+    CPU_CFLAGS += -mno-branch-likely -mno-mips16 -mno-interlink-compressed -mplt -msym32 -fno-caller-saves -fomit-frame-pointer -mno-long-calls -fsee -finline
     CPU_CFLAGS_mips32 = -mips32 -mtune=mips32
     CPU_CFLAGS_mips32r2 = -mips32r2 -mtune=mips32r2
     CPU_CFLAGS_mips64 = -mips64 -mtune=mips64 -mabi=64
