diff --git a/toolchain/gcc/Config.in b/toolchain/gcc/Config.in
index a8c39d6..cf68122 100644
--- a/toolchain/gcc/Config.in
+++ b/toolchain/gcc/Config.in
@@ -3,7 +3,7 @@
 choice
 	prompt "GCC compiler Version" if TOOLCHAINOPTS
 	default GCC_USE_VERSION_4_8_ARC if arc
-	default GCC_USE_VERSION_5
+	default GCC_USE_VERSION_5_2_LINARO
 	help
 	  Select the version of gcc you wish to use.
 
@@ -21,6 +21,11 @@ choice
 		select GCC_VERSION_5
 		bool "gcc 5.x"
 		depends on !arc
+		
+	config GCC_USE_VERSION_5_2_LINARO
+		select GCC_VERSION_5_2_LINARO
+		bool "gcc 5.2.x with Linaro enhancements"
+		depends on !arc
 
 endchoice
 
diff --git a/toolchain/gcc/common.mk b/toolchain/gcc/common.mk
index 3e4f3ee..aa38d5a 100644
--- a/toolchain/gcc/common.mk
+++ b/toolchain/gcc/common.mk
@@ -54,8 +54,15 @@ ifeq ($(findstring linaro, $(CONFIG_GCC_VERSION)),linaro)
       PKG_MD5SUM:=5ba2f3a449b1658ccc09d27cc7ab3c03
       PKG_COMP:=xz
     endif
+    ifeq ($(CONFIG_GCC_VERSION),"5.2-linaro")
+      PKG_REV:=5.2-2015.11
+      PKG_VERSION:=5.2.1
+      PKG_VERSION_MAJOR:=5.2
+      PKG_MD5SUM:=c65acd1d0fa4c6931a09c9c52c4cfa28
+      PKG_COMP:=xz
+    endif
     ifneq ($(LINARO_RELEASE),)
-      PKG_SOURCE_URL:=http://releases.linaro.org/$(LINARO_RELEASE)/components/toolchain/gcc-linaro/$(PKG_VERSION_MAJOR)
+      PKG_SOURCE_URL:=http://releases.linaro.org/components/toolchain/gcc-linaro/$(PKG_VERSION_MAJOR)
     else
       PKG_SOURCE_URL:=http://launchpad.net/gcc-linaro/$(PKG_VERSION_MAJOR)/$(PKG_REV)/+download/
     endif
diff --git a/toolchain/gcc/Config.version b/toolchain/gcc/Config.version
index efd8dc0..9f045a3 100644
--- a/toolchain/gcc/Config.version
+++ b/toolchain/gcc/Config.version
@@ -9,6 +9,7 @@ config GCC_VERSION
 	string
 	default "4.8-linaro"    if GCC_VERSION_4_8_LINARO
 	default "arc-2015.06"   if GCC_VERSION_4_8_ARC
+	default "5.2-linaro"    if GCC_VERSION_5_2_LINARO
 	default "5.2.0"
 
 config GCC_VERSION_4_8
@@ -18,3 +19,6 @@ config GCC_VERSION_4_8
 config GCC_VERSION_5
 	default y if (!TOOLCHAINOPTS && !arc)
 	bool
+
+config GCC_VERSION_5_2_LINARO
+	bool
