diff --git a/package/utils/busybox/Config-defaults.in b/package/utils/busybox/Config-defaults.in
index 2689b4f..1103c67 100644
--- a/package/utils/busybox/Config-defaults.in
+++ b/package/utils/busybox/Config-defaults.in
@@ -240,7 +240,7 @@ config BUSYBOX_DEFAULT_FEATURE_EDITING_VI
 	default n
 config BUSYBOX_DEFAULT_FEATURE_EDITING_HISTORY
 	int
-	default 256
+	default 8
 config BUSYBOX_DEFAULT_FEATURE_EDITING_SAVEHISTORY
 	bool
 	default n
@@ -1386,13 +1386,13 @@ config BUSYBOX_DEFAULT_FDFORMAT
 	default n
 config BUSYBOX_DEFAULT_FDISK
 	bool
-	default n
+	default y
 config BUSYBOX_DEFAULT_FDISK_SUPPORT_LARGE_DISKS
 	bool
-	default n
+	default y
 config BUSYBOX_DEFAULT_FEATURE_FDISK_WRITABLE
 	bool
-	default n
+	default y
 config BUSYBOX_DEFAULT_FEATURE_AIX_LABEL
 	bool
 	default n
@@ -2238,7 +2238,7 @@ config BUSYBOX_DEFAULT_FEATURE_TRACEROUTE_SOURCE_ROUTE
 	default n
 config BUSYBOX_DEFAULT_FEATURE_TRACEROUTE_USE_ICMP
 	bool
-	default n
+	default y
 config BUSYBOX_DEFAULT_TUNCTL
 	bool
 	default n
@@ -2349,7 +2349,7 @@ config BUSYBOX_DEFAULT_FEATURE_REFORMIME_COMPAT
 	default n
 config BUSYBOX_DEFAULT_SENDMAIL
 	bool
-	default n
+	default y
 config BUSYBOX_DEFAULT_IOSTAT
 	bool
 	default n
@@ -2454,7 +2454,7 @@ config BUSYBOX_DEFAULT_FEATURE_PS_UNUSUAL_SYSTEMS
 	default n
 config BUSYBOX_DEFAULT_RENICE
 	bool
-	default n
+	default y
 config BUSYBOX_DEFAULT_BB_SYSCTL
 	bool
 	default y
