--- a/target/linux/ar71xx/files/drivers/net/ethernet/atheros/ag71xx/ag71xx.h
+++ b/target/linux/ar71xx/files/drivers/net/ethernet/atheros/ag71xx/ag71xx.h
@@ -358,7 +358,7 @@ ag71xx_ring_desc(struct ag71xx_ring *rin
 #define RX_STATUS_PR		BIT(0)	/* Packet Received */
 #define RX_STATUS_OF		BIT(2)	/* Rx Overflow */
 #define RX_STATUS_BE		BIT(3)	/* Bus Error */
-
+/*
 static inline void ag71xx_check_reg_offset(struct ag71xx *ag, unsigned reg)
 {
 	switch (reg) {
@@ -371,10 +371,10 @@ static inline void ag71xx_check_reg_offs
 		BUG();
 	}
 }
-
+*/
 static inline void ag71xx_wr(struct ag71xx *ag, unsigned reg, u32 value)
 {
-	ag71xx_check_reg_offset(ag, reg);
+//	ag71xx_check_reg_offset(ag, reg);
 
 	__raw_writel(value, ag->mac_base + reg);
 	/* flush write */
@@ -383,7 +383,7 @@ static inline void ag71xx_wr(struct ag71
 
 static inline u32 ag71xx_rr(struct ag71xx *ag, unsigned reg)
 {
-	ag71xx_check_reg_offset(ag, reg);
+//	ag71xx_check_reg_offset(ag, reg);
 
 	return __raw_readl(ag->mac_base + reg);
 }
@@ -392,7 +392,7 @@ static inline void ag71xx_sb(struct ag71
 {
 	void __iomem *r;
 
-	ag71xx_check_reg_offset(ag, reg);
+//	ag71xx_check_reg_offset(ag, reg);
 
 	r = ag->mac_base + reg;
 	__raw_writel(__raw_readl(r) | mask, r);
@@ -404,7 +404,7 @@ static inline void ag71xx_cb(struct ag71
 {
 	void __iomem *r;
 
-	ag71xx_check_reg_offset(ag, reg);
+//	ag71xx_check_reg_offset(ag, reg);
 
 	r = ag->mac_base + reg;
 	__raw_writel(__raw_readl(r) & ~mask, r);
